### PR TITLE
Dove: removing unused external dependencies

### PR DESCRIPTION
**Pull request by vladimirovmm**

### CODE REVIEW

File: dove/src/context.rs [hunk 1]
```diff
@@ -60,6 +60,8 @@ impl Context {
         } else {
             let index = Index::build(package_hash, self)?;
             index.store(&index_path)?;
+            index.remove_unused(self.project_dir.join(&self.manifest.layout.deps).as_path())?;
```
Comment (RIg410) [inline dove/src/context.rs:63]: index.remove_unused(&self.path_for(&self.manifest.layout.deps))?;

File: dove/src/index/mod.rs [hunk 2]
```diff
@@ -194,6 +194,44 @@ impl Index {
...
+        let deps: Vec<PathBuf> = self
+            .deps_roots
+            .iter()
+            .filter_map(|item| {
+                let path = PathBuf::from(item);
+                path.components()
+                    .enumerate()
+                    .filter_map(|(num, part)| {
+                        part.as_os_str().to_str().and_then(|n| {
+                            if n.len() > 4 && &n[..4] == "git_" {
```
Comment (RIg410) [inline dove/src/index/mod.rs:214]: starts_with("git_") ?

File: dove/src/index/mod.rs [hunk 3]
```diff
@@ -194,6 +194,44 @@ impl Index {
...
+                                Some(num)
+                            } else {
+                                None
+                            }
+                        })
+                    })
+                    .last()
+                    .map(|num| path.components().take(num + 1).collect())
```
Comment (RIg410) [inline dove/src/index/mod.rs:222]: What happens if the git project is not dove?
Please check it out.

**Review by RIg410 — CHANGES_REQUESTED** [review]

File: dove/src/context.rs (hunk 1 above)
Comment (vladimirovmm) [reply dove/src/context.rs]: Thanks. Fixed it

File: dove/src/index/mod.rs (hunk 2 above)
Comment (vladimirovmm) [reply dove/src/index/mod.rs]: @RIg410  Thanks. Fixed it

File: dove/tests/test_dependency.rs [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+    for v in 1..=2 {
+        let project_folder = create_project_for_test_dependency(
+            "project_dependency_with_git_tag",
+            None,
+            Some(&format!("v{}", v)),
+            None,
+        );
+
+        // project_folder/scripts/version.move
+        add_sctipt_getversion(&project_folder);
```
Comment (vitvakatu) [inline dove/tests/test_dependency.rs:21]:
```suggestion
        add_script_getversion(&project_folder);
```

File: dove/src/index/mod.rs [hunk 5]
```diff
@@ -194,6 +195,44 @@ impl Index {
...
+            })
+            .filter(|path: &PathBuf| path.starts_with(deps_path))
+            .collect();
+
+        for finded in deps_path
```
[the remark this answers is not in the document]
Comment (vladimirovmm) [reply dove/src/index/mod.rs]: @vitvakatu Thanks. Fixed it

**Review by RIg410 — APPROVED** [review]

**Pull request merged**